File: archive/legacy-scripts/fix_gemini.js
const fs = require('fs');
let code = fs.readFileSync('backend/services/ocr_service.js', 'utf8');

// Change default Gemini model
code = code.replace(/modelName = 'gemini-2\.5-flash'/g, "modelName = 'gemini-1.5-pro'");

// Fix the fallback logic
const oldLogic = "const nextModel = modelName === 'gemini-2.5-flash' ? 'gemini-3.5-flash' : 'gemini-2.5-flash';";
const newLogic = "const nextModel = modelName === 'gemini-1.5-pro' ? 'gemini-1.5-flash' : 'gemini-2.5-flash';";
code = code.replace(oldLogic, newLogic);

// Fix the recursive fallback call
code = code.replace(/runGeminiVision\(processedPaths, 1, 'gemini-2\.5-flash'\)/g, "runGeminiVision(processedPaths, attempt + 1, nextModel)");

fs.writeFileSync('backend/services/ocr_service.js', code);
console.log("GEMINI FIXED");

File: archive/legacy-scripts/fix_gemini.py
import re

def fix_models(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Replace specific model strings with their -latest or -001 equivalents
    content = content.replace("'gemini-1.5-flash'", "'gemini-1.5-flash-latest'")
    content = content.replace("'gemini-1.5-pro'", "'gemini-1.5-pro-latest'")
    content = content.replace("'gemini-1.5-flash-8b'", "'gemini-1.0-pro-vision-latest'")

    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)

fix_models('backend/services/ocr_service.js')
fix_models('backend/services/rules_engine.js')
